src/BaseList/BaseList.tsx

```diff
@@ -0,0 +1,153 @@
+import clsx from "clsx";
+import React from "react";
+import {
+  ListContext,
+  ListGridContext,
+  useGridStyles,
+  useListContext,
+} from "./context";
+import { ListGridTemplate, useStyles } from "./styles";
+
+export interface BaseListProps extends React.HTMLAttributes<HTMLElement> {
+  gridTemplate: ListGridTemplate;
+}
+export const BaseList: React.FC<BaseListProps> = ({
+  children,
+  gridTemplate,
+  ...props
+}) => (
+  <ListGridContext.Provider value={gridTemplate}>
+    <section {...props}>{children}</section>
+  </ListGridContext.Provider>
+);
+
+export type BaseListItemClassKey =
+  | "row"
+  | "rowBody"
+  | "rowHead"
+  | "rowFoot"
+  | "rowHover"
+  | "rowBodySelected";
+
+export interface BaseListItemProps
+  extends React.HTMLAttributes<HTMLDivElement | HTMLLIElement> {
+  classes: Record<BaseListItemClassKey, string>;
+  hover?: boolean;
+  selected?: boolean;
+}
+
+export const BaseListItem: React.FC<BaseListItemProps> = ({
+  classes,
+  className,
```

Comment:
Why add `classes` and `className` separately and not add a key e.g `container` to `classes`?

Reply:
Because it's native; Material-UI does the same thing. `className` works as you'd expect - adds class name to most outer component, leaving `classes` to do tinkering with further customization. Also note that `classes` is omitted in `List` and `OffsettedList` components, leaving `className` as the only way to apply styles.